File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.js
function solution(n, t, m, p) {
    
    const len =  t*m;
    var str = "";
    var num = 0;
    
    while(str.length < len){
        str += num.toString(n);
        num++;
    }
    
    str = str.toUpperCase();
    var answer = '';
    var tube = p-1;
    
    for(let i=0; i<t; i++){
        answer += str.charAt(tube);
        tube += m;
    }
    
    return answer;
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String solution(int n, int t, int m, int p) {
        
        int len = t*m;
        String str = "";
        int num = 0;
        
        while(str.length() < len){
            String convert = Integer.toString(num, n);
            str += convert;
            num++;
        }
        
        str = str.toUpperCase();
        String answer = "";
        int tube = p-1;
        
        for(int i=0; i<t; i++){
            answer += str.charAt(tube);
            tube += m;
        }
        
        
        return answer;
    }
}